metrics_reporter.go

```diff
@@ -14,37 +14,48 @@ import (
 var (
 	percentiles     = []float64{0.75, 0.95, 0.99}
 	percentileNames = []string{"p75", "p95", "p99"}
-	lastCounts      = make(map[string]int64)
 )
 
 // Given a t representing a time in ns, convert to seconds, show up to μs precision
 func sec(t float64) string {
 	return fmt.Sprintf("%.6f", t/1000000000)
 }
 
-func countDifference(ctx slog.Context, name string, c int64) {
+type Emitter struct {
+	registry metrics.Registry
+	source   string
+	duration time.Duration
+	logger   *log.Logger
+	lastCounts map[string]int64
+}
+
+func NewEmitter(r metrics.Registry, source string, d time.Duration, l *log.Logger) *Emitter {
+	return &Emitter{registry: r, source: source, duration: d, logger: l, lastCounts: make(map[string]int64)}
+}
+
+func (e Emitter) countDifference(ctx slog.Context, name string, c int64) {
 	name = name + ".count"
-	lc := lastCounts[name]
+	lc := e.lastCounts[name]
 	ctx[name] = c - lc
-	lastCounts[name] = c
+	e.lastCounts[name] = c
 }
 
 // LogFmtMetricsEmitter emits the metrics in logfmt compatible formats every d
 // duration using the provided logger. source is added to the line as
 // log_shuttle_stats_source if not empty.
-func LogFmtMetricsEmitter(r metrics.Registry, source string, d time.Duration, l *log.Logger) {
-	if d == 0 {
+func (e Emitter) Emit() {
```

Comment:
Please fix the comment above, it should start with `// Emit ....`, as per normal Go conventions.